### ISSUE TITLE
Add validation preventing System.AccessViolationException when using `CreateInput` on uninitialized views

### ISSUE THREAD

**Issue by KirillAldashkin** [dotnet/Silk.NET#591]
# Summary
An `System.AccessViolationException` is thrown when creating an input
# Reproducing
- Platform: Windows 10
- Framework: .NET 5
## Steps
1) Create empty `Console application` project
2) Install packages:
```
Silk.NET.Input.Common 2.7.0
Silk.NET.Input.Desktop 1.9.0
Silk.NET.Input.Glfw 2.7.0
Silk.NET.Windowing.Common 2.7.0
Silk.NET.Windowing.Desktop 1.9.0
Silk.NET.Windowing 2.7.0
```
3) Paste this to `Program.cs` file:
```CSharp
using Silk.NET.Input;
using Silk.NET.Windowing;

var API = new GraphicsAPI(ContextAPI.OpenGLES,
    ContextProfile.Core,
    ContextFlags.Default,
    new APIVersion(2, 0));

var opts = ViewOptions.Default;
opts.FramesPerSecond = 50;
opts.UpdatesPerSecond = 50;
opts.API = API;
opts.ShouldSwapAutomatically = false;
opts.VSync = false;

var _window = Window.IsViewOnly ?
    Window.GetView(opts) :
    Window.Create(new(opts));

var input = _window.CreateInput();

_window.Run();
```
4) Build and run
## Stack trace
```
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at Silk.NET.GLFW.Glfw.SetCharCallback(Silk.NET.GLFW.WindowHandle*, CharCallback)
   at Silk.NET.Input.Glfw.GlfwEvents..ctor(Silk.NET.GLFW.WindowHandle*)
   at Silk.NET.Input.Glfw.GlfwInputPlatform.RegisterWindow(Silk.NET.GLFW.WindowHandle*, System.Collections.Generic.IEnumerable`1<Silk.NET.Input.Glfw.IGlfwSubscriber>)
   at Silk.NET.Input.Glfw.GlfwInputContext..ctor(Silk.NET.Windowing.IView)
   at Silk.NET.Input.Glfw.GlfwInputPlatform.CreateInput(Silk.NET.Windowing.IView)
   at Silk.NET.Input.InputWindowExtensions.CreateInput(Silk.NET.Windowing.IView)
   at <Program>$.<Main>$(System.String[])
```

**Comment by Perksey** [comment]
This is incorrect code as you need the window to be active to create an input context i.e. the window must already be running. Generally window contexts are created in the Load callback. However, this should not fail out in this way. How I think this would look:
- [ ] Add a `IsInitialized` property to `IView` (this is breaking, but we have said that additive breaking changes to HLU interfaces are fine so long as they're minor, in which case they are held back to 3.0)
- [ ] In `ViewImplementationBase`, make the existing property with the same name `public` with `protected set`
- [ ] In `InputWindowExtensions.CreateInput`, make it throw an exception if this property returns `false`.

Community contributions welcome.

**Comment by HurricanKai** [comment]
Do we need a new API for this? This is expected behaviour and is what happens if you try to do literally anything using an uninitialized window.

**Comment by Perksey** [comment]
_We_ should never cause an access violation, ever. If we do, this needs to be fixed to throw a cleaner, catchable exception. Currently, there is no way to detect an uninitialized window so that needs to be rectified.

**Comment by HurricanKai** [comment]
I suppose changing the exception type isn't breaking if the previous type wasn't catchable

**Comment by Perksey** [comment]
This has been fixed in #596 and will be in the 2.8 update this Friday.
